swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/BillableUsageAggregateConsumer.java

```diff
@@ -240,10 +242,13 @@ public AzureUsageContext lookupAzureUsageContext(BillableUsageAggregate billable
         if (isRecentlyTerminatedError) {
           throw new SubscriptionRecentlyTerminatedException(e);
         }
-        var isSubscriptionCannotBeDeterminedError =
+        var isNotFound =
             optionalErrors.get().getErrors().stream()
-                .anyMatch(error -> ("SUBSCRIPTIONS1006").equals(error.getCode()));
-        if (isSubscriptionCannotBeDeterminedError) {
+                .anyMatch(
+                    error ->
+                        (Integer.toString(Status.NOT_FOUND.getStatusCode()))
+                            .equals(error.getStatus()));
+        if (isNotFound) {
```

Comment:
I've checked that doing `Status.NOT_FOUND.getStatusCode() == Status.NOT_FOUND.getStae.getResponse().getStatus()` would also work (don't need to parse the integer to string and also loop over the list of errors. 

This is a minor suggestion, feel free to change it or not.